dashboard/src/components/providers/AppProvider.tsx

```diff
@@ -424,6 +424,13 @@ const middlewareReducer = async (
         `,
         );
       } catch (e: any) {
+        middlewareReducer(config, dispatch, {
```

Comment:
why is this not a dispatch call?

Reply:
In the `AppProvider` we have a `middleware` where we do the complex logic + API calls and then that's transferred to the `reducer` where we just make the transformations needed to the state and return. If we are inside the `middleware` a `dispatch` call will continue directly to the `reducer` but in this case we want to pass though the `middleware` again to make the API call to actually disconnect.